nursery/chain.go

```diff
@@ -0,0 +1,231 @@
+package nursery
+
+import (
+	"errors"
+	"fmt"
+	"github.com/BoltzExchange/boltz-client/boltz"
+	"github.com/BoltzExchange/boltz-client/boltzrpc"
+	"github.com/BoltzExchange/boltz-client/database"
+	"github.com/BoltzExchange/boltz-client/logger"
+)
+
+func (nursery *Nursery) sendChainSwapUpdate(swap database.ChainSwap) {
+	isFinal := swap.State == boltzrpc.SwapState_SUCCESSFUL || swap.State == boltzrpc.SwapState_REFUNDED
+	if swap.FromData.LockupTransactionId == "" && swap.State != boltzrpc.SwapState_PENDING {
+		isFinal = false
+	}
+
+	nursery.sendUpdate(swap.Id, SwapUpdate{
+		ChainSwap: &swap,
+		IsFinal:   isFinal,
+	})
+}
+
+func (nursery *Nursery) RegisterChainSwap(swap database.ChainSwap) error {
+	if err := nursery.registerSwap(swap.Id); err != nil {
+		return err
+	}
+	nursery.sendChainSwapUpdate(swap)
+	return nil
+}
+
+func (nursery *Nursery) setChainSwapLockupTransaction(swap *database.ChainSwap, data *database.ChainSwapData, transactionId string) error {
+	data.LockupTransactionId = transactionId
+	_, _, value, err := nursery.findLockupVout(data)
+	if err != nil {
+		return fmt.Errorf("could not find lockup vout: %s", err)
+	}
+
+	logger.Info("Got lockup transaction of Swap " + data.Id + " from Boltz: " + transactionId)
```

Comment:
One thing I am wondering about when looking at these logs: should we prefix those `Swap` with the type? Like `Chain Swap`? That's how I do it in the backend and that makes it quite a bit easier to follow along when multiple swaps are going on